### ISSUE TITLE
root being translated into newData.parent().parent() which does not work

### ISSUE THREAD

**Issue by mmniazi** [firebase/bolt#195]
Hey,
I am using bolt compiler with following rules:
```
path /users/{id} {
  read() { true }
  write() { isCurrentUser(id) || isOrganizationAdmin(id) }
}

isCurrentUser(id) {
    auth != null && auth.uid == id
}

isOrganizationAdmin(id) {
    isPartOfOrganization(id, getAdminOrganization())
}

isPartOfOrganization(id, organization) {
    root.users[id].organizations[organization] != null
}

getAdminOrganization() {
    root.admins[auth.uid].organization
}
```
This is translated to this:
```
{
  "rules": {
    "users": {
      "$id": {
        ".read": "true",
        ".write": "auth != null && auth.uid == $id || 
    newData.parent().parent().child('users').child($id).child('organizations').child(
        newData.parent().parent().child('admins').child(auth.uid).child('organization').val()
    ).val() != null"
      }
    }
  }
}
```

Problem is above rule does not work because `root` is translated into `newData.parent().parent()` but if I replace `newData.parent().parent()` with `root` it works in simulator, edited rules: 
```
{
  "rules": {
    "users": {
      "$id": {
        ".read": "true",
        ".write": "auth != null && auth.uid == $id || 
    root.child('users').child($id).child('organizations').child(
        newData.parent().parent().child('admins').child(auth.uid).child('organization').val()
    ).val() != null"
      }
    }
  }
}
```
I am using following sample data:
```
{
  "admins" : {
    "2YsitkLrhfSz41CPbjBQ3mEICo32" : {
      "organization" : "s2ditkLrhfSz41CPbjBQ3mEasd32"
    }
  },
  "users" : {
    "4KvfijVuY2R9PK5YzsHTfG2MDO03" : {
      "name" : "Mohsin",
      "organizations" : {
        "s2ditkLrhfSz41CPbjBQ3mEasd32" : {
          "name" : "Teamo"
        }
      }
    }
  }
}
```

This is how I am testing it:
![screenshot from 2017-06-02 09 57 18](https://cloud.githubusercontent.com/assets/7944966/26711477/f5db6380-4779-11e7-9df1-0f2e338bc8fa.png)

I hope that is enough to reproduce the issue. Let me know If I can help in any other way including contributions.

Thanks

**Comment by bpinney** [comment]
Have you tried using `prior(root)`:

```
path /users/{id} {
  read() { true }
  write() { isCurrentUser(id) || isOrganizationAdmin(id) }
}

isCurrentUser(id) {
    auth != null && auth.uid == id
}

isOrganizationAdmin(id) {
    isPartOfOrganization(id, getAdminOrganization())
}

isPartOfOrganization(id, organization) {
    prior(root).users[id].organizations[organization] != null
}

getAdminOrganization() {
    prior(root).admins[auth.uid].organization
}
```

which compiles to the following: 
```json
{
  "rules": {
    "users": {
      "$id": {
        ".read": "true",
        ".write": "auth != null && auth.uid == $id || root.child('users').child($id).child('organizations').child(root.child('admins').child(auth.uid).child('organization').val()).val() != null"
      }
    }
  }
}
```

**Comment by jacwright** [comment]
`prior(root)` works, but that seems a bit silly. Shouldn't `root` always compile to `root`?

**Comment by bpinney** [comment]
@mckoss [addressed](https://github.com/firebase/bolt/issues/87#issuecomment-160248217) this in the closed issue #87
